File: D.S.A/hashing/Find first non-repeating element.js
let arr=[1,1,2,2,3];

let map = new Map();
for(let num of arr){


    map.set(num,(map.get(num)||0)+1)
}

for(let [key,value] of map){
    if(value===1){
        console.log(key);
        break;
    }
}

File: Java/dsa/arrays/easy/hashing/FindFirstNonRepeatingElement.java
import java.util.LinkedHashMap;
import java.util.Map;

public class FindFirstNonRepeatingElement {
    public static Integer firstNonRepeating(int[] arr) {
        Map<Integer, Integer> map = new LinkedHashMap<>();
        for (int num : arr) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            if (entry.getValue() == 1) {
                return entry.getKey();
            }
        }
        return null;
    }
}
